frames/src/main/java/com/checkout/frames/screen/billingaddress/billingaddressdetails/BillingAddressDetailsViewModel.kt

```diff
@@ -0,0 +1,133 @@
+    private val buttonStateMapper: Mapper<ButtonStyle, InternalButtonState>,
+    private val style: BillingAddressDetailsStyle
+) : ViewModel() {
+    // Header styling
+    val screenTitleStyle = textLabelStyleMapper.map(style.billingAddressHeaderComponentStyle.headerTitleStyle)
+    val screenTitleState = provideScreenTitleState(style.billingAddressHeaderComponentStyle.headerTitleStyle)
+    val screenModifier = containerMapper.map(style.containerStyle)
+
+    val screenButtonState = provideButtonState()
+    val screenButtonStyle = buttonStyleMapper.map(style.billingAddressHeaderComponentStyle.headerButtonStyle)
+
+    val goBack = mutableStateOf(false)
+
+    val billingAddressInputComponentsContainerModifier =
+        containerMapper.map(style.billingAddressInputComponentsContainerStyle.containerStyle)
+
+    val billingAddressInputComponentState = provideBillingAddressInputComponentsState(
+        style.billingAddressInputComponentsContainerStyle
+    )
+    val billingAddressInputComponentsStyle = provideBillingAddressInputComponentsStyle(
+        style.billingAddressInputComponentsContainerStyle
+    )
+
+    private fun provideBillingAddressInputComponentsStyle(
+        billingAddressInputComponentsContainerStyle: BillingAddressInputComponentsContainerStyle
+    ): List<BillingAddressInputComponentViewStyle> {
+        return billingAddressDetailsComponentStyleUseCase.execute(
+            billingAddressInputComponentsContainerStyle.billingAddressInputComponentStyleList
+        )
+    }
+
+    private fun provideBillingAddressInputComponentsState(
+        billingAddressInputComponentsContainerStyle:
+        BillingAddressInputComponentsContainerStyle
+    ): BillingAddressInputComponentsContainerState {
+        return billingAddressDetailsComponentStateUseCase.execute(
+            billingAddressInputComponentsContainerStyle.billingAddressInputComponentStyleList
+        )
+    }
+
+    fun onFocusChanged(isFocused: Boolean) {
+        Log.d("", "" + isFocused)
+    }
+
+    fun onSearchChange(text: String) {
+        Log.d("", "" + text)
+    }
+
+    private fun provideButtonState(): InternalButtonState {
+        val state = buttonStateMapper.map(style.billingAddressHeaderComponentStyle.headerButtonStyle)
+        state.textState.textId.value = R.string.cko_billing_form_button_save
+        return state
+    }
+
+    private fun provideScreenTitleState(style: TextLabelStyle): TextLabelState {
+        val state = textLabelStateMapper.map(style)
+        state.textId.value = R.string.cko_billing_form_screen_title
+        state.leadingIcon.value = imageMapper.map(
+            ImageStyleToDynamicImageRequest(
+                style.leadingIconStyle,
+                flowOf(R.drawable.cko_ic_cross_close),
+                flowOf { goBack.value = true }
+            )
+        )
+
+        return state
+    }
+
+    fun onTapDoneButton() {
+        goBack.value = true
```

Comment:
On done button pressed address should be saved to the payment state manager and only after it closed.

Reply:
Sure, will do in the next PR.